### PR TITLE
Add Pub/Sub workflow tutorial

### PR DESCRIPTION
**Pull request by dwatrous**
This tutorial walks a user through creating topics, subscriptions and using a containerized app to process work associated with messages

### CODE REVIEW

File: tutorials/pubsub-workflow/index.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+```
+gcloud projects add-iam-policy-binding $PROJECT_ID --member=serviceAccount:$SERVICE_ACCOUNT@$PROJECT_ID.iam.gserviceaccount.com --role=roles/pubsub.publisher
+gcloud projects add-iam-policy-binding $PROJECT_ID --member=serviceAccount:$SERVICE_ACCOUNT@$PROJECT_ID.iam.gserviceaccount.com --role=roles/pubsub.subscriber
+gcloud projects add-iam-policy-binding $PROJECT_ID --member=serviceAccount:$SERVICE_ACCOUNT@$PROJECT_ID.iam.gserviceaccount.com --role=roles/storage.objectAdmin
+gcloud projects add-iam-policy-binding $PROJECT_ID --member=serviceAccount:$SERVICE_ACCOUNT@$PROJECT_ID.iam.gserviceaccount.com --role=roles/serviceusage.serviceUsageConsumer
+```
+
+## Create some work files
+The following commands will create a folder and fill it with 99 work files. Each file will containe up to 2048 bytes and will include specific tokens that we want to replace as part of our workflow.
```
Comment (complexsplit) [inline tutorials/pubsub-workflow/index.md:140]: containe typo.

File: tutorials/pubsub-workflow/index.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+export KEYFILE=$SERVICE_ACCOUNT-key.json
+export GOOGLE_APPLICATION_CREDENTIALS=$KEYFILE
+export DEMOSOURCE=/path/to/community/tutorials/pubsub-workflow/
+```
+
+### Where to run this tutorial
+This tutorial can be executed in any environment that provides gcloud and Docker. Some options include:
+
+ * [Cloud Shell](https://cloud.google.com/shell) includes Docker and has gcloud built in and ready to go.
+ * If you prefer to develop on your own host, install [Docker](https://www.docker.com/) and run the commands locally
```
Comment (complexsplit) [inline tutorials/pubsub-workflow/index.md:83]: Add period.

File: tutorials/pubsub-workflow/index.md (hunk 1 above)
Comment (dwatrous) [reply tutorials/pubsub-workflow/index.md]: fixed

File: tutorials/pubsub-workflow/index.md (hunk 2 above)
Comment (dwatrous) [reply tutorials/pubsub-workflow/index.md]: added

**Review by ToddKopriva — APPROVED** [review]

**Comment by ToddKopriva** [comment]
published:
https://cloud.google.com/community/tutorials/pubsub-workflow

**Pull request merged**